yt_dlp/extractor/platzi.py

```diff
@@ -111,17 +111,16 @@ def _real_extract(self, url):
         duration = video_player.get('duration', '')
         servers = try_get(video_player, lambda x: x['video']['servers'])
         formats = []
-        for server in servers.keys():
-            server_json = servers.get(server, {})
-            if 'hls' in server_json.keys():
+        for _, server_json in servers.items():
+            if server_json.get('hls'):
                 formats.extend(self._extract_m3u8_formats(
-                    server_json['hls'], lecture_id, 'mp4',
+                    server_json.get('hls'), lecture_id, 'mp4',
```

Comment:
I intentionally left this as `server_json['hls']` since this code will be reached only when the key is present